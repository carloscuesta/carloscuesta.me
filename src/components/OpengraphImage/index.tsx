type Props = {
  url?: string
  description: string
  title: string
  withProfile?: boolean
}

const OpengraphImage = (props: Props) => (
  <div
    style={{
      backgroundColor: 'white',
      width: '100%',
      height: '100%',
      display: 'flex',
      position: 'relative',
      fontFamily: 'Inter',
    }}
  >
    <div
      style={{
        display: 'flex',
        flexDirection: 'row',
        gap: '1em',
        alignItems: 'center',
        position: 'absolute',
        top: 60,
        left: 60,
      }}
    >
      <img
        src="https://carloscuesta.me/images/carloscuesta.jpg"
        width="100"
        height="100"
        style={{ borderRadius: '100%' }}
      />
      {props.withProfile && (
        <div style={{ display: 'flex', flexDirection: 'column' }}>
          <span
            style={{
              fontWeight: 700,
              fontSize: '2.75em',
              letterSpacing: '-1px',
            }}
          >
            Carlos Cuesta
          </span>
          <span
            style={{
              fontWeight: 300,
              fontSize: '1.75em',
            }}
          >
            @crloscuesta
          </span>
        </div>
      )}
    </div>

    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: '0.5rem',
        position: 'absolute',
        bottom: 140,
        left: 60,
        width: '65%',
      }}
    >
      <span
        style={{
          fontSize: '5.25rem',
          fontWeight: 700,
          letterSpacing: '-2px',
          lineHeight: '1',
        }}
      >
        {props.title}
      </span>
      <span
        style={{
          fontSize: '2.5rem',
          fontWeight: 600,
        }}
      >
        {props.description}
      </span>
    </div>
    <div
      style={{
        position: 'absolute',
        bottom: 60,
        left: 60,
        fontSize: '1.75rem',
        lineHeight: 1,
        fontWeight: 600,
        borderRadius: 100,
        fontFamily: 'RobotoMono',
      }}
    >
      {props.url ? `carloscuesta.me/${props.url}` : 'carloscuesta.me'}
    </div>
  </div>
)

export default OpengraphImage
