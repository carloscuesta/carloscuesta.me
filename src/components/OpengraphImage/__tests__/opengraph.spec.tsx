import { render } from '@testing-library/react'
import fetch, { enableFetchMocks } from 'jest-fetch-mock'

import OpengraphImage, { getFonts } from '..'

describe('OpengraphImage', () => {
  it('should match a page opengraph', () => {
    const { container } = render(
      <OpengraphImage
        title="Blog"
        description="This is a test description"
        url="/"
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should match a blog post opengraph', () => {
    const { container } = render(
      <OpengraphImage
        title="Writing ADRs"
        description="This is a test description"
        url="/blog"
        isBlogPost
      />,
    )

    expect(container).toMatchSnapshot()
  })

  describe('getFonts', () => {
    beforeAll(() => {
      enableFetchMocks()
    })

    it('should return an array of fonts', async () => {
      const fonts = await getFonts()

      expect(fonts).toMatchSnapshot()
    })
  })
})
