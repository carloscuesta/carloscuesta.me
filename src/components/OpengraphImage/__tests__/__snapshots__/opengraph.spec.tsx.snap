// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpengraphImage getFonts should return an array of fonts 1`] = `
[
  {
    "data": ArrayBuffer [],
    "name": "Inter",
    "style": "normal",
    "weight": 600,
  },
  {
    "data": ArrayBuffer [],
    "name": "Inter",
    "style": "normal",
    "weight": 700,
  },
  {
    "data": ArrayBuffer [],
    "name": "RobotoMono",
    "style": "normal",
    "weight": 300,
  },
]
`;

exports[`OpengraphImage should match a blog post opengraph 1`] = `
<div>
  <div
    style="background-color: white; width: 100%; height: 100%; display: flex; flex-direction: column; position: relative; font-family: Inter; padding: 60px; justify-content: space-between; background-size: 20px 20px;"
  >
    <div
      style="display: flex; flex-direction: column; justify-content: space-between; flex: 1; margin-bottom: 3rem;"
    >
      <div
        style="display: flex; flex-direction: row; gap: 1em; align-items: center;"
      >
        <img
          height="115"
          src="https://carloscuesta.me/images/carloscuesta.jpg"
          style="border-radius: 100%;"
          width="115"
        />
        <div
          style="display: flex; flex-direction: column;"
        >
          <span
            style="font-weight: 700; font-size: 3em; letter-spacing: -1px;"
          >
            Carlos Cuesta
          </span>
          <span
            style="font-weight: 300; font-size: 1.85em;"
          >
            @crloscuesta
          </span>
        </div>
      </div>
      <div
        style="display: flex; flex-direction: column; gap: 1.25rem;"
      >
        <div
          style="font-size: 4.5rem; font-weight: 700; letter-spacing: -2px; line-height: 1;"
        >
          Writing ADRs
        </div>
        <div
          style="font-size: 2.5rem;"
        >
          This is a test description
        </div>
      </div>
    </div>
    <div
      style="font-size: 1.75rem; line-height: 1; font-weight: 600; font-family: RobotoMono;"
    >
      carloscuesta.me/blog
    </div>
  </div>
</div>
`;

exports[`OpengraphImage should match a page opengraph 1`] = `
<div>
  <div
    style="background-color: white; width: 100%; height: 100%; display: flex; position: relative; font-family: Inter; background-size: 20px 20px;"
  >
    <img
      height="115"
      src="https://carloscuesta.me/images/carloscuesta.jpg"
      style="border-radius: 100%; position: absolute; left: 60px; top: 60px;"
      width="115"
    />
    <div
      style="display: flex; flex-direction: column; gap: 0.5rem; position: absolute; bottom: 140px; left: 60px; width: 65%;"
    >
      <span
        style="font-size: 5.25rem; font-weight: 700; letter-spacing: -2px; line-height: 1;"
      >
        Blog
      </span>
      <span
        style="font-size: 2.5rem; font-weight: 600;"
      >
        This is a test description
      </span>
    </div>
    <div
      style="position: absolute; bottom: 60px; left: 60px; font-size: 1.75rem; line-height: 1; font-weight: 600; border-radius: 100px; font-family: RobotoMono;"
    >
      carloscuesta.me//
    </div>
  </div>
</div>
`;
